api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/pvls/PregnantCalculation.java

```diff
@@ -0,0 +1,217 @@
+ *
+ * @return CalculationResultMap
+ */
+@Component
+public class PregnantCalculation extends AbstractPatientCalculation {
+
+  @Autowired private HivMetadata hivMetadata;
+
+  @Override
+  public CalculationResultMap evaluate(
+      Collection<Integer> cohort,
+      Map<String, Object> parameterValues,
+      PatientCalculationContext context) {
+
+    CalculationResultMap resultMap = new CalculationResultMap();
+
+    Location location = (Location) context.getFromCache("location");
+    Date oneYearBefore = EptsCalculationUtils.addMonths(context.getNow(), -12);
+
+    EncounterType labEncounterType = hivMetadata.getMisauLaboratorioEncounterType();
+    EncounterType adultFollowup = hivMetadata.getAdultoSeguimentoEncounterType();
+    EncounterType pediatriaFollowup = hivMetadata.getARVPediatriaSeguimentoEncounterType();
+
+    Concept viralLoadConcept = hivMetadata.getHivViralLoadConcept();
+    Concept pregnant = hivMetadata.getPregnantConcept();
+    Concept yes = hivMetadata.getYesConcept();
+    Concept pregnantBasedOnWeeks = hivMetadata.getNumberOfWeeksPregnant();
+    Concept pregnancyDueDate = hivMetadata.getPregnancyDueDate();
+    Program ptv = hivMetadata.getPtvEtvProgram();
+
+    // get female patients only
+    Set<Integer> femaleCohort = EptsCalculationUtils.female(cohort, context);
+
+    CalculationResultMap pregnantMap =
+        EptsCalculations.getObs(
+            pregnant,
+            femaleCohort,
+            Arrays.asList(location),
+            Arrays.asList(yes),
+            TimeQualifier.ANY,
+            null,
+            context);
+
+    CalculationResultMap markedPregnantByWeeks =
+        EptsCalculations.getObs(
+            pregnantBasedOnWeeks,
+            femaleCohort,
+            Arrays.asList(location),
+            null,
+            TimeQualifier.ANY,
+            null,
+            context);
+
+    CalculationResultMap markedPregnantDueDate =
+        EptsCalculations.getObs(
+            pregnancyDueDate,
+            femaleCohort,
+            Arrays.asList(location),
+            null,
+            TimeQualifier.ANY,
+            null,
+            context);
+
+    CalculationResultMap markedPregnantInProgram =
+        EptsCalculations.lastProgramEnrollment(ptv, femaleCohort, context);
+
+    CalculationResultMap lastVl =
+        EptsCalculations.lastObs(
+            Arrays.asList(labEncounterType, adultFollowup, pediatriaFollowup),
+            viralLoadConcept,
+            location,
+            oneYearBefore,
+            context.getNow(),
+            cohort,
+            context);
+
+    Set<Integer> onArtForMoreThan3Months =
+        EptsCalculationUtils.patientsThatPass(
+            calculate(
+                Context.getRegisteredComponents(OnArtForMoreThanXmonthsCalcultion.class).get(0),
+                femaleCohort,
+                context));
+
+    for (Integer pId : femaleCohort) {
+
+      Boolean isCandidate = false;
+      if (onArtForMoreThan3Months.contains(pId)) {
```

Comment:
I wonder why the patient needs to be on ART for more than 3 months to be considered pregnant. 
This calculation will only be valid for `TX_PVLS` right? 
It can stay the way it is, but would it be better to have this condition on the composition cohort?

Reply:
@ynurmahomed, **good question**.I realized that the verification of  on ART for more than 3 months is made in others cohorts (other parts of the code)  and when needed the result is crossed with the pregnant/breastfeeding result, I removed this verification and the result remain the same, I will remove this verification

Reply:
:+1: